python/pyspark/sql/functions.py

```diff
@@ -2184,6 +2184,11 @@ def pandas_udf(f=None, returnType=None, functionType=None):
        |         8|      JOHN DOE|          22|
        +----------+--------------+------------+
 
+       .. note:: The length of `pandas.Series` within a scalar UDF is not of the whole input column
+           but of the batch internally used, and it is called for each batch. Therefore,
```

Comment:
Does this sound a little better?  "..scalar UDF is not that of the whole input column, but is the length of an internal batch used for each call to the function."

Reply:
Yup, English isn't really my area :(. Will try to incorporate your suggestion.